src/main/java/org/snomed/aag/data/domain/CriteriaItemSignOffFactory.java

```diff
@@ -0,0 +1,32 @@
+package org.snomed.aag.data.domain;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class CriteriaItemSignOffFactory {
+	/**
+	 * Return CriteriaItemSignOff. If branch is the same as ProjectAcceptanceCriteria.branchPath, CriteriaItemSignOffProject will be returned. Otherwise, CriteriaItemSignOffTask
+	 * will be returned.
+	 *
+	 * @param criteriaItemId            Value for CriteriaItemSignOff.
+	 * @param branch                    Value for CriteriaItemSignOff.
+	 * @param branchHeadTimestamp       Value for CriteriaItemSignOff.
+	 * @param projectIteration          Optional value for CriteriaItemSignOff.
+	 * @param userId                    Value for CriteriaItemSignOff.
+	 * @param projectAcceptanceCriteria Required for determining type of CriteriaItemSignOff
+	 * @return CriteriaItemSignOff
+	 */
+	public CriteriaItemSignOff create(String criteriaItemId, String branch, Long branchHeadTimestamp, Integer projectIteration, String userId, ProjectAcceptanceCriteria projectAcceptanceCriteria) {
+		boolean branchProjectLevel = projectAcceptanceCriteria.isBranchProjectLevel(branch);
+		if (branchProjectLevel) {
+			if (projectIteration == null || projectIteration < 0) {
```

Comment:
Nice validation, do we also need this in the find criteria items methods?

Reply:
There are similar validation methods to verify the parameters passed to the findX methods, .e.g 

```
if (criteriaItemId == null || branchPath == null || (projectIteration != null && projectIteration < 0) || projectAcceptanceCriteria == null) {
throw new IllegalArgumentException(INVALID_PARAMETERS);
}
```